tests/abi/tools/revng-abi-verify/ABIArtifactParser.cpp

```diff
@@ -0,0 +1,188 @@
+/// \file Verify.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+
+#include "revng/ABI/FunctionType.h"
+#include "revng/Model/Binary.h"
+
+#include "ABIArtifactParser.h"
+
+namespace State {
+
+using namespace abi::artifact;
+
+struct SingleRun {
+  llvm::StringRef Function;
+  bool IsLittleEndian;
+  Registers Registers;
+  Stack Stack;
+  Arguments ReturnValue;
+  Arguments Arguments;
+};
+using Deserialized = llvm::SmallVector<SingleRun, 16>;
+
+} // namespace State
+
+//
+// `llvm::yaml` traits for the internal state.
+//
+
+template<>
+struct llvm::yaml::MappingTraits<State::Register> {
+  static void mapping(IO &IO, State::Register &N) {
+    IO.mapRequired("Name", N.Name);
+    IO.mapRequired("Value", N.Value);
+  }
+};
```

Comment:
Mostly because I never actually touched it. Writing a straightforward mapping seemed faster to me at a time, especially considering that this yaml is pretty simple.

Either way, I have nothing against moving to `TupleTreeGenerator`.

Reply:
Let's keep it as it is for now.
Things are moving in other PRs on the CMake side of `TupleTreeGenerator`.
Please leave a TODO about this.

Reply:
Sure.

Reply:
Done: https://github.com/revng/revng/pull/148/commits/cd1439dfa5a530eedd9c0fbaa674f2406750e83a#diff-1ff4591728310ca98928087735ded8f7d2f661d82ccb9668a4e41fb01e4ee09fR33.